backend/app/adapter/db/user_changelog.go

```diff
@@ -0,0 +1,85 @@
+package db
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/short-d/short/app/adapter/db/table"
+	"github.com/short-d/short/app/entity"
+	"github.com/short-d/short/app/usecase/repository"
+)
+
+var _ repository.UserChangeLog = (*UserChangeLogSQL)(nil)
+
+type UserChangeLogSQL struct {
+	db *sql.DB
+}
+
+func (u UserChangeLogSQL) GetLastViewedAt(user entity.User) (time.Time, error) {
+	statement := fmt.Sprintf(`
+SELECT "%s" 
+FROM "%s"
+WHERE "%s"=$1;`,
+		table.UserChangeLog.ColumnLastViewedAt,
+		table.UserChangeLog.TableName,
+		table.UserChangeLog.ColumnEmail,
+	)
+
+	row := u.db.QueryRow(statement, user.Email)
+	lastViewedAt := time.Time{}
+	err := row.Scan(&lastViewedAt)
+
+	return lastViewedAt, err
+}
+
+func (u UserChangeLogSQL) UpdateLastViewedAt(user entity.User, currentTime time.Time) (time.Time, error) {
+	lastViewedAt, err := u.GetLastViewedAt(user)
+	if err != nil {
+		return lastViewedAt, err
+	}
```

Comment:
```suggestion
```

Reply:
@byliuyang  If we remove this we don't get an error if updation fails to find user in user_changelog

Reply:
@kapillamba4 Interesting. Maybe let use case handle user not found error?

Reply:
Not sure about this one. Will look back in the future.

Reply:
@byliuyang  All other places where updation is happening we are returning an error if entry is not found: https://github.com/short-d/short/blob/master/backend/app/adapter/db/user.go#L170-L173

Reply:
@kapillamba4 Okey. Sounds good!